ladybug/src/main/java/nl/nn/ibistesttool/filter/BlackBox.java

```diff
@@ -15,53 +15,31 @@
 */
 package nl.nn.ibistesttool.filter;
 
-import java.util.List;
-
 import nl.nn.testtool.Checkpoint;
-import nl.nn.testtool.Report;
-import nl.nn.testtool.filter.CheckpointMatcher;
 
 /**
  * @author Jaco de Groot
  */
-public class BlackBox implements CheckpointMatcher {
-	
-	public boolean match(Report report, Checkpoint checkpoint) {
-		if (
-				checkpoint.getName() != null
-				&&
-				(
-					checkpoint.getName().startsWith("Sender ")
-					// Also in stub4testtool.xsl
-					&& !"nl.nn.adapterframework.jdbc.DirectQuerySender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.jdbc.FixedQuerySender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.DelaySender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.EchoSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.IbisLocalSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.LogSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.ParallelSenders".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.SenderSeries".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.SenderWrapper".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.XsltSender".equals(checkpoint.getSourceClassName())
-					// Not in stub4testtool.xsl
-					&& !"nl.nn.adapterframework.senders.FixedResultSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.XmlValidatorSender".equals(checkpoint.getSourceClassName())
-				)
-			) {
+public class BlackBox extends GrayBox {
+
+	@Override
+	protected boolean match(Checkpoint checkpoint) {
+		if (checkpoint.getName() != null && checkpoint.getName().startsWith("Sender ")
+				// Also in stub4testtool.xsl
+				&& !"nl.nn.adapterframework.jdbc.DirectQuerySender".equals(checkpoint.getSourceClassName())
```

Comment:
I would suggest making a `Set<String>` of all class-names enumerated here, then checking if `checkpoint.getSourceClassName()` is not in this set.

Reply:
Done